lms/djangoapps/courseware/user_state_client.py

```diff
@@ -193,7 +154,10 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
         else:
             user = User.objects.get(username=username)
 
-        evt_time = time()
+        if user.is_anonymous():
+            # Anonymous users cannot be persisted to the database, so let's just use
+            # what we have.
+            return
```

Comment:
@jcdyer Let's keep the Datadog metrics in place.  No point in removing it since our performance team thinks the data will continue to be useful.